operators/utilities.py

```diff
@@ -38,6 +38,52 @@ def select_all(context: Context):
 def deselect_all(context: Context):
     global_data.selected.clear()
 
+def select_invert(context: Context):
+    sketch = context.scene.sketcher.active_sketch
+    if sketch:
+        generator = sketch.sketch_entities(context)
+    else:
+        generator = entities_3d(context)
+    
+    for e in generator:
+        e.selected = not e.selected
+
+def select_extend(context: Context):
+    sketch = context.scene.sketcher.active_sketch
+    if sketch:
+        generator = sketch.sketch_entities(context)
+    else:
+        generator = entities_3d(context)
+    
+    to_select = []
+    for e in generator:
+        if e.is_segment():
+            if e.selected:
+                if hasattr(e, 'p1'):
+                    to_select.append(e.p1)
+                if hasattr(e, 'p2'):
+                    to_select.append(e.p2)
+                continue
+            if (hasattr(e,'p1') and e.p1.selected) or (hasattr(e, 'p2') and e.p2.selected):
```

Comment:
This should be an `elif` so you avoid adding elements that are already selected.

Reply:
Done 👍